python/pyarrow/_exec_plan.pyx

```diff
@@ -235,163 +237,121 @@ def _perform_join(join_type, left_operand not None, left_keys,
-            # so that we can coalesce the keys and exclude duplicates in a subsequent projection.
-            left_columns_set = set(left_columns)
-            right_columns_set = set(right_columns)
-            # Where the right table columns start.
-            right_operand_index = len(left_columns)
-            for idx, col in enumerate(left_columns + right_columns):
-                if idx < len(left_columns) and col in left_column_keys_indices:
-                    # Include keys only once and coalesce left+right table keys.
-                    c_projected_col_names.push_back(tobytes(col))
-                    # Get the index of the right key that is being paired
-                    # with this left key. We do so by retrieving the name
-                    # of the right key that is in the same position in the provided keys
-                    # and then looking up the index for that name in the right table.
-                    right_key_index = right_column_keys_indices[right_keys[left_keys_order[col]]]
-                    c_projections.push_back(Expression.unwrap(
-                        Expression._call("coalesce", [
-                            Expression._field(idx), Expression._field(
-                                right_operand_index+right_key_index)
-                        ])
-                    ))
-                elif idx >= right_operand_index and col in right_column_keys_indices:
-                    # Do not include right table keys. As they would lead to duplicated keys.
-                    continue
-                else:
-                    # For all the other columns incude them as they are.
-                    # Just recompute the suffixes that the join produced as the projection
-                    # would lose them otherwise.
-                    if left_suffix and idx < right_operand_index and col in right_columns_set:
-                        col += left_suffix
-                    if right_suffix and idx >= right_operand_index and col in left_columns_set:
-                        col += right_suffix
-                    c_projected_col_names.push_back(tobytes(col))
-                    c_projections.push_back(
-                        Expression.unwrap(Expression._field(idx)))
-            c_decl_plan.push_back(
-                CDeclaration(tobytes("project"), CProjectNodeOptions(
-                    c_projections, c_projected_col_names))
-            )
-    else:
-        c_decl_plan.push_back(
-            CDeclaration(tobytes("hashjoin"), CHashJoinNodeOptions(
-                c_join_type, c_left_keys, c_right_keys,
-                _true,
-                <c_string>tobytes(left_suffix or ""),
-                <c_string>tobytes(right_suffix or "")
-            ))
+    # TEMP
+    if isinstance(left_operand, Dataset):
+        left_operand = left_operand.to_table()
+    if isinstance(right_operand, Dataset):
+        right_operand = right_operand.to_table()
```

Comment:
This is just a temporary hack because for now there is only a TableSourceNodeOptions, I need to add a proper ScanNodeOptions in another pre-cursor PR